File: backend/src/main/java/com/tripfriend/domain/qna/dto/AnswerDto.java
package com.tripfriend.domain.qna.dto;

import com.tripfriend.domain.qna.entity.Answer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class AnswerDto {
    private Long answerId;
    private String content;
    private String memberUsername;
    private String createdAt;

    public AnswerDto(Answer answer) {
        this.answerId = answer.getId();
        this.content = answer.getContent();
        this.memberUsername = answer.getMember().getUsername();
        this.createdAt = answer.getCreatedAt().toString();
    }
}

File: backend/src/main/java/com/tripfriend/domain/qna/dto/AnswerDto.kt
package com.tripfriend.domain.qna.dto

import com.tripfriend.domain.qna.entity.Answer

data class AnswerDto(
    val answerId: Long?,
    val content: String,
    val memberUsername: String,
    val createdAt: String
) {
    constructor(answer: Answer) : this(
        answerId = answer.id,
        content = answer.content,
        memberUsername = answer.member.username,
        createdAt = answer.createdAt.toString()
    )
}